scripts/pdf/render.py

```diff
@@ -15,106 +15,66 @@
 import argparse
 from datetime import datetime
 
-from weasyprint import HTML
-
+from weasyprint import HTML, CSS
 
 def main(loc, colorscheme):
-
     oslist = []
     allmd = []
-    group = []
-    ap = []
 
     # Checking correctness of path
     if not os.path.isdir(loc):
         print("Invalid directory. Please try again!", file=sys.stderr)
         sys.exit(1)
 
-    # Writing names of all directories inside 'pages' to a list
-    for os_dir in os.listdir(loc):
-        oslist.append(os_dir)
-
-    oslist.sort()
-
-    # Required strings to create intermediate HTML files
-    header = '<!doctype html><html><head><meta charset="utf-8"><link rel="stylesheet" href="basic.css">'
+    # Set up css style sheets
+    csslist = [CSS("basic.css")]
     if colorscheme != "basic":
-        header += '<link rel="stylesheet" href="' + colorscheme + '.css"></head><body>\n'
+        csslist.append(CSS(colorscheme + ".css"))
 
-    header += "</head><body>\n"
-    footer = "</body></html>"
-    title_content = "<h1 class=title-main>tldr pages</h1>" \
+    # A string that stores all pages in HTML format
+    html = '<!doctype html><html><head><meta charset="utf-8">' \
+        +"</head><body><h1 class=title-main>tldr pages</h1>" \
         + "<h4 class=title-sub>Simplified and community-driven man pages</h4>" \
         + "<h6 class=title-sub><em><small>Generated on " + datetime.now().strftime("%c") + "</small></em></h6>" \
-        + "</body></html>"
-
-    # Creating title page
-    with open("title.html", "w") as f:
-        f.write(header + title_content)
+        + '<p style="page-break-before: always" ></p>'
 
-    group.append(HTML("title.html").render())
+    # Writing names of all directories inside 'pages' to a list
+    for os_dir in os.listdir(loc):
+        oslist.append(os_dir)
+    oslist.sort()
 
     for operating_sys in oslist:
```

Comment:
Can combine these into a single for loop:

```suggestion
    for operating_sys in sorted(os.listdir(loc)):
```